src/transaction/mvcc_table.cpp

```diff
@@ -298,14 +298,18 @@ mvcctable::build_mvcc_info (log_tdes &tdes)
       trans_status.m_active_mvccs.copy_to (tdes.mvccinfo.snapshot.m_active_mvccs,
 					   mvcc_active_tran::copy_safety::THREAD_UNSAFE);
       /* load statistics temporary disabled need to be enabled when activate count optimization */
-#if 0
       /* load global statistics. This must take place here and nowhere else. */
-      if (logtb_load_global_statistics_to_tran (thread_p) != NO_ERROR)
+      tdes.mvccinfo.snapshot.valid = true;
```

Comment:
This flag is supposed to be set below after initializing all variables of `tdes.mvccinfo.snapshot`. I don't know in details, but I concern that there could be a race condition in which someone access uninitialized variables after checking this flag. Could you tell me the specific code building mvcc info recursively in `logtb_load_global_statistics_to_tran()`?

Reply:
The setting code is originally located furthur down, however, if the valid flage is not set here, this mvcc build routine can be called endless recursively by partitioned table. And there ar no setting conditions as you know.

Reply:
There is no memory barrier around it, so I can guess it is not referred to by others as you said. Then, we can remove `tdes.mvccinfo.snapshot.valid = true` below, but because of code readability, it still would be better to leave as it is.

Reply:
Ok, I will leave as it is.

Reply:
Ok, I will do to leave as it is.

Reply:
I mean that we should find a way not to set true here. I'm leaving a comment to avoid miscommunication just in case.

Reply:
I misunderstood what you said, but I couldn't find a way to set the flag true else here.

Reply:
So, I'd like to know the exact point calling it recursively in gtb_load_global_statistics_to_tran().

Reply:
In partitioned table, the call stack is as below:
```
logtb_tran_load_global_stats_func
	partition_load_pruning_context
		heap_get_class_partitions
			heap_class_get_partition_info
				logtb_get_mvcc_snapshot
					logtb_tran_load_global_stats_func
```
endless recursve call stack is made because mvcc_snapshot.valid flag is not set true neverthless mvcc info lis loading yet.
If we should modify the mechanism, we should add a flag named "loading" as boolean type. The flag must be set before calling  logtb_tran_load_global_stats_func and reset it after the function call is done.

Reply:
Thank you for the information :). The way I see it, `loading` or 'building' flag for a new state could be a quick solution. @hornetmj What do you think about it?

Reply:
We discussed it and with further analysis, it turns out that we don't need to build mvcc info in `heap_class_get_partition_info()`. We can revert the change related to mvcc build and remove mvcc building in the function to get rid of the infinite recursive calling.

Reply:
logtb_get_mvcc_snapshot at heap_class_get_partition_info calling at was removed not to make endleess recursive call.